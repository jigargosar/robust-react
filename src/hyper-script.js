import h from 'react-hyperscript'
import hh from 'hyperscript-helpers'

export const {div} = hh(h)


export {h}
