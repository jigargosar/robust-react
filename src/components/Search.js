import {connect} from '@cerebral/react'
import {signal, state} from 'cerebral/tags'
import {TextField, withStyles} from 'material-ui'
import {h} from '../hyper-script'

const Search = withStyles(({spacing: {unit}}) => ({
  root: {
    backgroundColor: 'rgba(255,255,255,0.1)',
    color: 'inherit',
  },
  input: {
    margin: `0 ${unit}px`,
  },
}))(function Search({searchText, onSearchInputChange, classes}) {
  return h(TextField, {
    value: searchText,
    onChange: onSearchInputChange,
    type: 'search',
    placeholder: 'Search',
    InputProps: {
      disableUnderline: true,
      classes: {
        root: classes.root,
        input: classes.input,
      },
    },
  })
})

export default connect(
  {
    setSearchText: signal`setSearchText`,
    searchText: state`searchText`,
  },
  ({setSearchText, searchText}) => ({
    onSearchInputChange: event =>
      setSearchText({searchText: event.target.value}),
    searchText,
  }),
  Search,
)
