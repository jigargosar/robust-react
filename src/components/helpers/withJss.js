import {createGenerateClassName} from 'material-ui/styles'
import JssProvider from 'react-jss/lib/JssProvider'
/* eslint-disable babel/new-cap */
import {render} from 'react-testing-library'
import {h} from '../../hyper-script'
import S from '../../sanctuary'

const generateClassName1 = (rule, styleSheet) =>
  `${styleSheet.options.classNamePrefix}-${rule.key}`

const generateClassName2 = createGenerateClassName({
  dangerouslyUseGlobalCSS: true,
})

const generateClassName = S.I(true) ? generateClassName1 : generateClassName2
const withJss = component =>
  render(
    h(JssProvider, {generateClassName}, component),
    // component,
  )
export default withJss
