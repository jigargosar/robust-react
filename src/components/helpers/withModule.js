import {Container} from '@cerebral/react'
import {Controller} from 'cerebral'
import {Module} from 'cerebral/index'
import {h} from '../../hyper-script'

const withModule = (module, component) =>
  h(Container, {controller: Controller(Module(module))}, [component])

export default withModule
