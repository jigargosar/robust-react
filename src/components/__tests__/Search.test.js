import {h} from '../../hyper-script'
import withJss from '../helpers/withJss'
import withModule from '../helpers/withModule'
import Search from '../Search'

it('renders without crashing', () => {
  const {container} = withJss(withModule({}, h(Search)))
  expect(container.firstChild).toMatchSnapshot()
})
