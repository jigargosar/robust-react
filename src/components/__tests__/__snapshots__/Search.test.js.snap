// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  class="MuiFormControl-root"
>
  <div
    class="MuiInput-root Component-root MuiInput-formControl"
  >
    <input
      aria-invalid="false"
      aria-required="false"
      class="MuiInput-input Component-input MuiInput-inputType MuiInput-inputTypeSearch"
      placeholder="Search"
      type="search"
      value=""
    />
  </div>
</div>
`;
